src/libs/hooks/useScrollLock/helpers/handleOverflow.js

```diff
@@ -0,0 +1,43 @@
+let previousPaddingRight;
+let previousOverflowSetting;
+
+function hiddenOverflow() {
+    setTimeout(() => {
+        if (previousPaddingRight === undefined) {
+            const scrollBarGap = window.innerWidth - document.documentElement.clientWidth;
+
+            if (scrollBarGap > 0) {
+                previousPaddingRight = document.body.style.paddingRight;
+                document.body.style.paddingRight = `${scrollBarGap}px`;
+            }
+        }
+
+        if (previousOverflowSetting === undefined) {
+            previousOverflowSetting = document.body.style.overflow;
+            document.body.style.overflow = 'hidden';
+        }
+    });
+}
+
+function restoreOverflow() {
+    setTimeout(() => {
+        if (previousPaddingRight !== undefined) {
+            document.body.style.paddingRight = previousPaddingRight;
+            previousPaddingRight = undefined;
+        }
+
+        if (previousOverflowSetting !== undefined) {
+            document.body.style.overflow = previousOverflowSetting;
+            previousOverflowSetting = undefined;
+        }
+    });
+}
+
+export default function handleOverflow(locks) {
+    if (locks.length === 1) {
+        hiddenOverflow();
+    }
+    if (locks.length === 0) {
+        restoreOverflow();
+    }
```

Comment:
what happens when locks have more than 1 element?

Reply:
When the length of lock is 1 the scroll will become lock, when it is greater than 1 it is not necessary to do anything because it is already lock.